src/goose/synopsis/process_manager.py

```diff
@@ -0,0 +1,100 @@
+import subprocess
+import os
+from typing import Literal, Dict
+from rich.markdown import Markdown
+from rich.rule import Rule
+from goose.notifier import Notifier
+from goose.synopsis.system import system
+from goose.toolkit.utils import RULEPREFIX, RULESTYLE
+from goose.utils.shell import is_dangerous_command, keep_unsafe_command_prompt
+
+ProcessManagerCommand = Literal["start", "list", "view_output", "cancel"]
+
+
+class ProcessManager:
+    def __init__(self, notifier: Notifier) -> None:
+        self.notifier = notifier
+
+        # Command dispatch dictionary
+        self.command_dispatch = {
+            "start": self._start_process,
+            "list": self._list_processes,
+            "view_output": self._view_process_output,
+            "cancel": self._cancel_process,
+        }
+
+    def logshell(self, command: str, title: str = "shell") -> None:
+        self.notifier.log("")
+        self.notifier.log(
+            Rule(RULEPREFIX + f"{title} | [dim magenta]{os.path.abspath(system.cwd)}[/]", style=RULESTYLE, align="left")
+        )
+        self.notifier.log(Markdown(f"```bash\n{command}\n```"))
+        self.notifier.log("")
+
+    def _start_process(self, shell_command: str, **kwargs: dict) -> int:
+        """Start a background process running the specified command."""
+        self.logshell(shell_command, title="background")
+
+        if is_dangerous_command(shell_command):
+            self.notifier.stop()
+            if not keep_unsafe_command_prompt(shell_command):
+                raise RuntimeError(f"The command {shell_command} was rejected as dangerous.")
+            self.notifier.start()
+
+        process = subprocess.Popen(
+            shell_command,
+            shell=True,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            text=True,
+            cwd=system.cwd,
+            env=system.env,
+        )
+        process_id = system.add_process(process)
+        return process_id
+
+    def _list_processes(self, **kwargs: dict) -> Dict[int, str]:
+        """List all running background processes."""
+        processes = system.get_processes()
+        process_list = "```\n" + "\n".join(f"id: {pid}, command: {cmd}" for pid, cmd in processes.items()) + "\n```"
+        self.notifier.log("")
+        self.notifier.log(Rule(RULEPREFIX + "processes", style=RULESTYLE, align="left"))
+        self.notifier.log(Markdown(process_list))
+        self.notifier.log("")
+        return processes
+
+    def _view_process_output(self, process_id: int, **kwargs: dict) -> str:
+        """View the output of a running background process."""
+        self.notifier.log("")
+        self.notifier.log(Rule(RULEPREFIX + "processes", style=RULESTYLE, align="left"))
+        self.notifier.log(Markdown(f"```\nreading {process_id}\n```"))
+        self.notifier.log("")
+        output = system.view_process_output(process_id)
```

Comment:
out of scope for this PR but we should test this but i believe in the event a process failed we don't catch `stderr` so that would be important to capture (which may aid in system recovery).